core/src/main/java/io/grpc/inprocess/InProcessServerBuilder.java

```diff
@@ -98,10 +105,31 @@ private InProcessServerBuilder(String name) {
     handshakeTimeout(Long.MAX_VALUE, TimeUnit.SECONDS);
   }
 
+  /**
+   * Provides a custom scheduled executor service.
+   *
+   * <p>It's an optional parameter. If the user has not provided a scheduled executor service when
+   * the channel is built, the builder will use a static cached thread pool.
+   *
+   * @return this
+   *
+   * @since 1.11.0
+   */
+  public InProcessServerBuilder scheduledExecutorService(
+      ScheduledExecutorService scheduledExecutorService) {
+    this.scheduledExecutorService =
+        checkNotNull(scheduledExecutorService, "scheduledExecutorService");
+    return this;
+  }
+
   @Override
   protected InProcessServer buildTransportServer(
       List<ServerStreamTracer.Factory> streamTracerFactories) {
-    return new InProcessServer(name, GrpcUtil.TIMER_SERVICE, streamTracerFactories);
+    ObjectPool<ScheduledExecutorService> schedulerPool =
+        scheduledExecutorService == null
+            ? SharedResourcePool.forResource(GrpcUtil.TIMER_SERVICE)
```

Comment:
Use a ObjectPool field instead of ScheduledExecutorService and make this the default value. Then in the `scheduledExecutorService()` method overwrite it with the FixedObjectPool.

You could do this pattern in InProcessChannelBuilder as well.

Reply:
Fixed. The pattern does not provide much gain for InProcessChannelBuilder, and currently it's in the same pattern as NettyChannelBuilder and OkHttpChannelBuilder.